crates/matrix-sdk-crypto/src/verification/sas/mod.rs

```diff
@@ -940,31 +941,31 @@ mod tests {
         let (content, request_info) =
             bob.receive_any_event(alice.user_id(), &content.into()).unwrap();
         assert!(!bob.can_be_presented());
-        matches!(bob.state(), SasState::Accepted { .. });
+        assert_matches!(bob.state(), SasState::Accepted { .. });
         bob.mark_request_as_sent(&request_info.unwrap().request_id);
 
         assert!(bob.can_be_presented());
-        matches!(bob.state(), SasState::KeysExchanged { .. });
+        assert_matches!(bob.state(), SasState::KeysExchanged { .. });
 
         let content = KeyContent::try_from(&content).unwrap();
         alice.receive_any_event(bob.user_id(), &content.into());
-        matches!(alice.state(), SasState::KeysExchanged { .. });
+        assert_matches!(alice.state(), SasState::KeysExchanged { .. });
         assert!(alice.can_be_presented());
 
         assert_eq!(alice.emoji().unwrap(), bob.emoji().unwrap());
         assert_eq!(alice.decimals().unwrap(), bob.decimals().unwrap());
 
         let mut requests = alice.confirm().await.unwrap().0;
-        matches!(alice.state(), SasState::Confirmed);
+        assert_matches!(alice.state(), SasState::Confirmed);
         assert!(requests.len() == 1);
         let request = requests.pop().unwrap();
         let content = OutgoingContent::try_from(request).unwrap();
         let content = MacContent::try_from(&content).unwrap();
         bob.receive_any_event(alice.user_id(), &content.into());
-        matches!(bob.state(), SasState::KeysExchanged { .. });
+        assert_matches!(bob.state(), SasState::KeysExchanged { .. });
 
         let mut requests = bob.confirm().await.unwrap().0;
-        matches!(bob.state(), SasState::Confirmed);
+        assert_matches!(bob.state(), SasState::Done { .. });
```

Comment:
We have been using matches! without assertion here, meaning the outcome was not actually tested. As a result the state at this specific point is already Done instead of Confirmed, presumably because the second confirm moves to done automatically.

Reply:
Thanks for catching this :sweat_smile: